### PR TITLE
CSPL-834 Refine Ingress Documentation

### PR DESCRIPTION
**Pull request by vebken-splunk**
CSPL-834 Refine Ingress Documentation
Clarify that we need to install a custom inputs.conf to open 9998 on the indexers.

Clarify that we will explain how to configure Splunk Web and HEC in the istio section.

CSPL-896 uploading larger TA results in nginx 413 error

Add relevant nginx proxy annotations to our examples so that users are less likely to run into nginx 413 errors.

CSPL-831 Include ingress-nginx Splunk Web example on Ingress documentation.

Include ingress nginx splunk web tls example in the ingress documentation and clarify nginxinc tls configuration.

### CODE REVIEW

File: docs/Ingress.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ For instructions on how to install and configure Istio for your specific infrast
 
 Most scenarios for Istio will require the configuration of a Gateway and a Virtual Service. Familiarize yourself with the [Istio Gateway](https://istio.io/latest/docs/reference/config/networking/gateway/) and [Istio Virtual Service ](https://istio.io/latest/docs/reference/config/networking/virtual-service/).
 
-### Configuring Ingress for Splunk Web
+### Configuring Ingress for Splunk Web and HEC
```
Comment (smohan-splunk) [inline docs/Ingress.md:56]: I see we have updated the heading to include HEC, but the sub-points under the topic have no examples for HEC?

Comment (vebken-splunk) [reply docs/Ingress.md:56]: Under the "Multiple Hosts Configuration" sub-point we reference HEC, 
`If your deployment has multiple hosts such as Search Heads and Cluster Master, use this example to configure Splunk Web access, and HTTP Event Collector port. Follow the steps here [HEC Documentation](https://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector) to learn how to create a HEC token and how to send data using HTTP.`
perhaps I could update that sub point heading to Multiple Hosts and HEC Configuration? Or perhaps we could include a solely HEC example with a standalone

Comment (smohan-splunk) [reply docs/Ingress.md:56]: Yes, at least an update to the existing Heading will be good!

**Review by smohan-splunk — APPROVED** [review]

**Review by mgaldino-splunk — APPROVED** [review]
LGTM.

File: docs/Ingress.md [hunk 2]
```diff
@@ -383,6 +383,8 @@ spec:
 
 If you only have one indexer cluster that you would like to use as the destination for all S2S traffic, you can optionally replace `splunk.example.com` in the above examples with the wildcard `*`. When you use this wildcard, you do not have to set the `tlsHostname` parameter in `outputs.conf` on your forwarders.
 
+Additionally, it is necessary to push an inputs.conf to both configure the relevant TLS settings and open port 9998 for receiving on the indexers as these configurations are not default. This can be achieved by installing an app containing a custom inputs.conf to the indexers through the Operator's [app management capabilities](https://splunk.github.io/splunk-operator/Examples.html#installing-splunk-apps).
```
Comment (eddikay) [inline docs/Ingress.md:386]: 4. Deploy an app to the standalone instance with the inputs.conf settings needed to open port 9998 and configure the relevant TLS settings. For details on app management using the Splunk Operator, see [Using Apps for Splunk Configuration](Examples.md#using-apps-for-splunk-configuration).

File: docs/Ingress.md [hunk 3]
```diff
@@ -543,6 +551,59 @@ spec:
           servicePort: 8000
 ```
 
+Example to create a TLS enabled Ingress configuration
```
Comment (eddikay) [inline docs/Ingress.md:554]: It's best not to leave any prerequisites, notes, conditions, or other gotchas at the bottom. 

Example to create a TLS enabled Ingress configuration
There are a few important configurations to be aware of in the TLS configuration:
* The `nginx.ingress.kubernetes.io/backend-protocol:` requires `"HTTPS"` when TLS is configured on the backend services.
* The `secretName` must reference a [valid TLS secret](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets).

Note: This example assumes that https is enabled for Splunk Web.

File: docs/Ingress.md [hunk 4]
```diff
@@ -543,6 +551,59 @@ spec:
...
+        backend:
+          serviceName: splunk-idc-indexer-service
+          servicePort: 8088
+  tls:
+  - hosts:
+    - shc.example.com
+    - hec.example.com
+    secretName: operator-tls
+```
+Note:
```
Comment (eddikay) [inline docs/Ingress.md:598]: See comment above about removing lines 598 - 605 and moving them above the example.

File: docs/Ingress.md [hunk 5]
```diff
@@ -705,6 +766,12 @@ status:
   loadBalancer: {}
 ```
 
+Note:
```
Comment (eddikay) [inline docs/Ingress.md:769]: See comment at line 554 about moving conditions above the example.

File: docs/Ingress.md (hunk 3 above)
Comment (vebken-splunk) [reply docs/Ingress.md:554]: Got it, will update.

File: docs/Ingress.md [hunk 6]
```diff
@@ -112,7 +112,7 @@ kubectl get svc -n istio-system
 http://<LoadBalance-External-IP>
```
[the remark this answers is not in the document]
Comment (vebken-splunk) [reply docs/Ingress.md]: Fixed

File: docs/Ingress.md (hunk 4 above)
Comment (vebken-splunk) [reply docs/Ingress.md]: Fixed

File: docs/Ingress.md (hunk 2 above)
Comment (vebken-splunk) [reply docs/Ingress.md]: Fixed

**Review by eddikay — APPROVED** [review]

**Pull request merged**